### PR TITLE
Add more details about retention to storage docs

### PR DESCRIPTION
**Pull request by SuperQ**
* Make compaction docs a little more clear, easy to find.
* Expand compaction docs slightly.
* Add notes about block cleanup to operational section.

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

File: docs/storage.md [hunk 1]
```diff
@@ -70,6 +75,8 @@ If your local storage becomes corrupted for whatever reason, your best bet is to
 
 If both time and size retention policies are specified, whichever policy triggers first will be used at that instant.
 
+Expired block cleanup happens on a background schedule, so it may take some time (hours) after startup to remove expired blocks. Expired blocks must be fully time expired before they are cleaned up.
```
Comment (brian-brazil) [inline docs/storage.md]: fully time expired reads oddly.

It'll happen on the next reload() ,which is usually every 2h.

File: docs/storage.md [hunk 2]
```diff
@@ -42,12 +42,17 @@ The directory structure of a Prometheus server's data directory will look someth
...
 
 Note that a limitation of the local storage is that it is not clustered or replicated. Thus, it is not arbitrarily scalable or durable in the face of disk or node outages and should thus be treated as more of an ephemeral sliding window of recent data. However, if your durability requirements are not strict, you may still succeed in storing up to years of data in the local storage.
 
 For further details on file format, see [TSDB format](https://github.com/prometheus/tsdb/blob/master/docs/format/README.md).
 
+## Compaction
+
+The initial two-hour blocks are eventually compacted into longer blocks in the background.
+
+Compaction will create larger blocks up to 10% of the rention time, or 21 days, which ever is smaller.
```
Comment (brian-brazil) [inline docs/storage.md]: whichever

File: docs/storage.md (hunk 1 above)
Comment (SuperQ) [reply docs/storage.md]: Yea, I wasn't happy with that wording either. Will fix it up.

**Comment by SuperQ** [comment]
@brian-brazil Made some adjustments to the wording.

**Pull request merged**